### PR TITLE
[celotool] use ensure0x/strip0x, fix sync-tests, generalize getValidators

### PR DESCRIPTION
**Pull request by mcortesi**
### Description

* use ensure0x/strip0x, remove duplicated logic
* refactor: generalize getValidator() 
* fix sync tests + clean e2e utils

### Tested

* `yarn lint-checks`
* Run geth_sync_test locally

### Notes

This is preparation work for celotool-faucet deployment

### CODE REVIEW

**Review by nambrot — DISMISSED** [review]

File: packages/celotool/src/lib/utils.ts [hunk 1]
```diff
@@ -300,3 +300,6 @@ export function addCeloGethMiddleware(argv: yargs.Argv) {
 export const validateAccountAddress = (address: string) => {
   return address !== null && address.toLowerCase().startsWith('0x') && address.length === 42 // 0x followed by 40 hex-chars
 }
+
+export const ensure0x = (hexstr: string) => (hexstr.startsWith('0x') ? hexstr : '0x' + hexstr)
+export const strip0x = (hexstr: string) => (hexstr.startsWith('0x') ? hexstr.slice(2) : hexstr)
```
Comment (trianglesphere) [inline packages/celotool/src/lib/utils.ts:304]: I've written a version of this, it's `stripHexLeader` in `utils/src/commentEncryption.ts`. Web3-utils also has a version of this. It'd be nice to use one set of functions for these conversions. I haven't seen another `ensure0x`  and I do like it.

**Review by nambrot — DISMISSED** [review]

File: packages/celotool/src/lib/utils.ts (hunk 1 above)
Comment (asaj) [reply packages/celotool/src/lib/utils.ts:304]: +1, we should at some point move these to `packages/utils` and use them everywhere

File: packages/celotool/src/lib/generate_utils.ts [hunk 2]
```diff
@@ -71,26 +71,31 @@ export const generateAccountAddressFromPrivateKey = (privateKey: string) => {
   return new Web3.modules.Eth().accounts.privateKeyToAccount(ensure0x(privateKey)).address
 }
 
+export const privateKey2GethAddress = (privateKey: string) =>
```
Comment (asaj) [inline packages/celotool/src/lib/generate_utils.ts]: nit: consider `privateKeyToStrippedAddress`, or `privateKeyToAddressNoPrefix`, or something that makes it clear that this is a normal address without the `0x` prefix

File: packages/celotool/src/lib/generate_utils.ts [hunk 3]
```diff
@@ -71,26 +71,31 @@ export const generateAccountAddressFromPrivateKey = (privateKey: string) => {
...
-export const getValidators = (mnemonic: string, n: number) => {
-  return range(0, n)
-    .map((i) => generatePrivateKey(mnemonic, AccountType.VALIDATOR, i))
-    .map(generateAccountAddressFromPrivateKey)
-    .map(strip0x)
-}
+export const getAddressesFor = (accountType: AccountType, mnemonic: string, n: number) =>
+  getPrivateKeysFor(accountType, mnemonic, n).map(generateAccountAddressFromPrivateKey)
+
+export const getGethAddressesFor = (accountType: AccountType, mnemonic: string, n: number) =>
```
Comment (asaj) [inline packages/celotool/src/lib/generate_utils.ts]: ditto

**Review by asaj — DISMISSED** [review]
Very nice!

File: packages/celotool/src/lib/utils.ts (hunk 1 above)
Comment (mcortesi) [reply packages/celotool/src/lib/utils.ts:304]: I agree, i would rather keep this PR constrained to celotool for now. And along this with function, i believe there are many others we can extract.

i don't if commentEncryption is the right module for...maybe `util/hex` or `util/addresses`

**Review by ashishb — DISMISSED** [review]

**Review by jmrossy — DISMISSED** [review]

**Review by jmrossy — APPROVED** [review]

**Pull request merged**